baggage/baggage_test.go

```diff
@@ -487,18 +494,25 @@ func TestBaggageString(t *testing.T) {
 		},
 		{
 			name: "URL encoded value",
-			out:  "foo=1%3D1",
+			out:  "foo=%20%22%2C%3B%5C%25%0D%0A%09",
```

Comment:
```suggestion
			// Allowed value characters are:
			//
			//   %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
			//
			// Meaning, US-ASCII characters excluding CTLs, whitespace,
			// DQUOTE, comma, semicolon, and backslash. All excluded
			// characters need to be percent encoded.
			out: "foo=%00%01%02%03%04%05%06%07%08%09%0A%0B%0C%0D%0E%0F%10%11%12%13%14%15%16%17%18%19%1A%1B%1C%1D%1E%1F%20!%22#$%&\()*+%2C-./0123456789:%3B<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[%5C]^_%60abcdefghijklmnopqrstuvwxyz{|}~%7F",
```

Reply:
If I checked correctly only backtick ``` (`%60`) is escaped but it does not have to be.

Reply:
PTAL f882da5d

Reply:
> 3\. Precent % must be escaped into %25.

Where does this come from? The baggage specification allows `%x25` from my reading:

> %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E

Reply:
> Now, the result is "correct", but not "perfect".

I'm not sure I agree, I don't think the result is correct. It is still escaping characters that are allowed by the baggage specification, right?

I think a correct solution will include correctly escaping the output in order for this PR to "Fix escaping in Member.String".

Reply:
> 1. Backtick ` (%60) does not have to be escaped

:+1:

Reply:
> 2\. It should be ' instead of \

:+1:

Reply:
> The baggage specification allows %x25 from my reading:

You are correct. I find it as a bug in the specification. Otherwise how would you know if a `%20` stands for ` ` or `%20`?
In my opinion this should be clarified here: https://www.w3.org/TR/baggage/#value. I will create a bug in https://github.com/w3c/baggage

Reference: https://datatracker.ietf.org/doc/html/rfc3986#section-2.1

Reply:
I created:
- https://github.com/w3c/baggage/issue/124
- https://github.com/w3c/baggage/pull/125

Reply:
> True, but this is spec-compliant. From: https://www.w3.org/TR/baggage/#value
> 
> > Characters which are not required to be percent-encoded MAY be percent-encoded.

Ah, gotcha. Yeah, a follow-up PR sounds appropriate than.